bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetThermoregulationHandler.java

```diff
@@ -0,0 +1,375 @@
+    }
+
+    private QuantityType<Temperature> commandToQuantityType(Command command, Unit<Temperature> unit) {
+        return new QuantityType<Temperature>(command.toFullString());
+    }
+
+    @Override
+    protected void handleMessage(BaseOpenMessage msg) {
+        super.handleMessage(msg);
+        if (msg.isCommand()) {
+            updateModeAndFunction((Thermoregulation) msg);
+        } else {
+            if (msg.getDim() == null) {
+                return;
+            }
+            if (msg.getDim() == Thermoregulation.DIM.TEMPERATURE
+                    || msg.getDim() == Thermoregulation.DIM.PROBE_TEMPERATURE) {
+                updateTemperature((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.TEMP_SETPOINT
+                    || msg.getDim() == Thermoregulation.DIM.COMPLETE_PROBE_STATUS) {
+                updateSetpoint((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.VALVES_STATUS) {
+                updateValveStatus((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.ACTUATOR_STATUS) {
+                updateActuatorStatus((Thermoregulation) msg);
+            } else if (msg.getDim() == Thermoregulation.DIM.FAN_COIL_SPEED) {
+                updateFanCoilSpeed((Thermoregulation) msg);
+            } else {
+                logger.debug("handleMessage() Ignoring unsupported DIM {} for thing {}. Frame={}", msg.getDim(),
+                        getThing().getUID(), msg);
+            }
+        }
+    }
+
+    private void updateModeAndFunction(Thermoregulation tmsg) {
+        logger.debug("updateModeAndFunction() for thing: {} msg={}", thing.getUID(), tmsg);
+
+        if (tmsg.getWhat() == null) {
+            logger.debug("updateModeAndFunction() Could not parse Mode from: {}", tmsg.getFrameValue());
+            return;
+        }
+
+        try {
+            logger.debug("updateModeAndFunction() WHAT={}", tmsg.getWhat());
+            WHAT w = WHAT.fromValue(tmsg.getWhat().value());
+
+            OPERATION_MODE mode = w.mode();
+            FUNCTION function = w.function();
+
+            if (w == WHAT.HEATING) {
+                function = FUNCTION.HEATING;
+            } else if (w == WHAT.CONDITIONING) {
+                function = FUNCTION.COOLING;
+            }
+
+            if (mode == OPERATION_MODE.MANUAL) {
+                logger.debug("updateModeAndFunction() function={} mode={} setPointTemp={}°", function.toString(),
+                        mode.toString(), currentSetPointTemp);
+            } else {
+                logger.debug("updateModeAndFunction() function={} mode={}", function.toString(), mode.toString());
+            }
+
+            updateState(CHANNEL_MODE, new StringType(mode.toString()));
+            updateState(CHANNEL_FUNCTION, new StringType(function.toString()));
+
+            // store current function
+            currentFunction = function;
+        } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type?

Reply:
I wanted to catch generic exception due to cast conversion or null values but I've removed the try{} catch{} and added a check on null values.